internal/commands/user/list/inputs.go

```diff
@@ -0,0 +1,27 @@
+package list
+
+import (
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+)
+
+type inputs struct {
+	cli.ProjectAppInputs
+	UserState     realm.UserState
+	Pending       bool
+	ProviderTypes []string
+	Users         []string
+}
+
+func (i *inputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if !areValidProviderTypes(i.ProviderTypes) {
```

Comment:
it's probably worth adding an `inputs_test.go` that makes sure provider types are resolved/validated properly.  that and/or just some unit tests in `flags_test.go` here could work